main.go

```diff
@@ -88,108 +96,175 @@ func (c *DownloadPlugin) Run(cliConnection plugin.CliConnection, args []string)
 		os.Exit(1)
 	}
 
-	flagVals := ParseFlags(args)
+	cmdExec := cmd_exec.NewCmdExec()
```

Comment:
Yes. The `dir_parser` on line 136 can also be shared by all the downloads so I'll move that initialization out of the for loop as well.